pkgs/by-name/ha/hare/hook.nix

```diff
@@ -0,0 +1,62 @@
+{
+  hare,
+  lib,
+  makeSetupHook,
+  makeWrapper,
+  runCommand,
+  stdenv,
+  writeShellApplication,
+}:
+let
+  arch = stdenv.targetPlatform.uname.processor;
+  harePropagationInputs = builtins.filter (
+    drv:
+    builtins.any (inputName: lib.getName drv == inputName) [
+      "harec"
+      "qbe"
+    ]
+  ) hare.nativeBuildInputs;
```

Comment:
it may be less fragile to expose these in hare's `passthru` attribute and then reference them here as `harePropagationInputs = [ hare.passthru.harec hare.passthru.qbe ];`